src/rosdistro/rosdistro.py

```diff
@@ -199,24 +199,21 @@ def _fetch_package_xml(self, rosdistro):
         repo = self.repository
         if 'github.com' in repo.url:
             url = repo.url
-            upstream_version = repo.version.split('-')[0]
-            release_tag = 'release/{0}/{1}'.format(self.name, upstream_version)
-            url = url.replace('.git', '/{0}/package.xml'.format(release_tag))
+            tail = 'release/{0}/{1}/{2}/package.xml'.format(rosdistro, self.name, repo.version)
```

Comment:
I would suggest to keep the variable name as `release_tag` since the string is the release tag.